MyOnboarding/src/MyOnboardingApp.Database/Repository/TodoListRepository.cs

```diff
@@ -1,49 +1,69 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Threading;
 using System.Threading.Tasks;
+using MongoDB.Driver;
+using MyOnboardingApp.Contracts.Database;
 using MyOnboardingApp.Contracts.Models;
 using MyOnboardingApp.Contracts.Repository;
 
 namespace MyOnboardingApp.Database.Repository
 {
     internal class TodoListRepository: ITodoListRepository
     {
-        private static readonly TodoListItem[] s_items = {
-            new TodoListItem
-            {
-                Text = "1st Todo Item",
-                Id = new Guid("11111111-1111-1111-1111-aabbccddeeff")
-            },
-            new TodoListItem
-            {
-                Text = "2nd Todo Item",
-                Id = new Guid("22222222-2222-2222-2222-aabbccddeeff")
-            },
-            new TodoListItem
-            {
-                Text = "3rd Todo Item",
-                Id = new Guid("33333333-3333-3333-3333-aabbccddeeff")
-            }
-        };
-
-
-        public async Task<IEnumerable<TodoListItem>> GetAllItemsAsync() 
-            => await Task.FromResult(s_items);
-        
+        private readonly IMongoDatabase _database;
+        private readonly IMongoCollection<TodoListItem> _databaseItems;
+        private const string CollectionName = "TodoItems";
+
+
+        public TodoListRepository(IDatabaseConnection connectionString)
+        {
+            _database = GetDatabaseFromConnection(connectionString);
+            _databaseItems = LoadCollectionFromDatabase(CollectionName);
+        }
+
+
+        public async Task<IEnumerable<TodoListItem>> GetAllItemsAsync()
+        {
+            var results = await _databaseItems.FindAsync(FilterDefinition<TodoListItem>.Empty);
```

Comment:
You can use following code here:
```suggestion
            var results = await _databaseItems.Find(FilterDefinition<TodoListItem>.Empty).ToListAsync();
```